knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEngine.java

```diff
@@ -17,6 +17,9 @@
  */
```

Comment:
There is a lot of functionalities here in ServiceEngine from now on, and unit test would be strongly required to assure it's working as expected.
I'd suggest at least to unit tests of ServiceEngine class